isofit/core/forward.py

```diff
@@ -362,21 +362,22 @@ def calibration(self, x):
 
     def upsample(self, wl, q):
         """Linear interpolation to RT wavelengths."""
-
-        # In some cases, these differ only by a tiny amount,
-        # so no need to waste time interpolating
-        if (len(wl) == len(self.RT.wl)) and np.allclose(wl, self.RT.wl):
-            return q
-
-        if q.ndim > 1:
-            q2 = []
-            for qi in q:
-                p = interp1d(wl, qi, fill_value="extrapolate")
-                q2.append(p(self.RT.wl))
-            return np.array(q2)
-        else:
-            p = interp1d(wl, q, fill_value="extrapolate")
-            return p(self.RT.wl)
+        # Only interpolate if these aren't close
+        close = len(wl) == len(self.RT.wl) and np.allclose(wl, self.RT.wl)
+
+        # or if any dimension is the wrong size
+        interp = (np.array(q.shape) == len(self.RT.wl)).all()
+
+        if not close or interp:
+            if q.ndim > 1:
+                for qi in q:
+                    p = interp1d(wl, qi, fill_value="extrapolate")
+                    q2.append(p(self.RT.wl))
+                return np.array(q2)
```

Comment:
@jammont Is `q2` a typo? It leads to failure of the checks.

Reply:
Accidentally missed the line `q2 = []` which is a temporary holder list. Fixed!